src/wp-includes/database-types.php

```diff
@@ -0,0 +1,182 @@
+<?php
+/**
+ * Database type API
+ *
+ * @package WordPress
+ * @subpackage Database type
+ * @since n.e.x.t
+ */
+
+/**
+ * Get supported types for metadata and option tables.
+ *
+ * @since n.e.x.t
+ *
+ * @return array List of supported database types.
+ */
+function wp_get_database_types() {
```

Comment:
The name of this function, and its related functions, is just a little confusing to me. Could something like `wp_get_meta_storage_types()` be a better choice?

(I realize this PR is still a draft—this is just a small spot of early feedback.)